lib/queue.js

```diff
@@ -26,16 +28,9 @@ module.exports = FetchQueue;
 
 FetchQueue.prototype = [];
 FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callback) {
```

Comment:
Since the `crawler.processURL` method basically does the same thing as the proposed `crawler.queue.createQueueItem`, this raises a good point about compatibility between the recurring but undocumented `parsedURL` type object and `queueItem`s.

I think what we ought to change here is to have `crawler.processURL` construct a full `queueItem` (which would mean adding the properties `url`, `depth`, `fetched`, `status` and `stateData` to the returned object) and then have the `add` method basically just push it to the queue. I was thinking about having the `add` method set a `queueItem.id` property as well, but considering the fact that URL's are unique, the `queueItem.url` property should be able to fill that function for queue implementations that need to look up the queue item in eg. a database.